tests/test_bot.py

```diff
@@ -1431,3 +1431,93 @@ def assertion(url, data, *args, **kwargs):
         monkeypatch.setattr(bot.request, 'post', assertion)
 
         assert bot.close()
+
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_copy_message(self, monkeypatch, bot, chat_id, media_message):
+        keyboard = InlineKeyboardMarkup(
+            [[InlineKeyboardButton(text="test", callback_data="test2")]]
+        )
+
+        def post(url, data, timeout):
+            assert data["chat_id"] == chat_id
+            assert data["from_chat_id"] == chat_id
+            assert data["message_id"] == media_message.message_id
+            assert data["caption"] == "<b>Test</b>"
+            assert data["parse_mode"] == ParseMode.HTML
+            assert data["reply_to_message_id"] == media_message.message_id
+            assert data["reply_markup"] == keyboard
+            return data
+
+        monkeypatch.setattr(bot.request, 'post', post)
+        bot.copy_message(
+            chat_id,
+            from_chat_id=chat_id,
+            message_id=media_message.message_id,
+            caption="<b>Test</b>",
+            parse_mode=ParseMode.HTML,
+            reply_to_message_id=media_message.message_id,
+            reply_markup=keyboard,
+        )
+
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_copy_message_without_reply(self, bot, chat_id, media_message):
+        keyboard = InlineKeyboardMarkup(
+            [[InlineKeyboardButton(text="test", callback_data="test2")]]
+        )
+
+        returned = bot.copy_message(
+            chat_id,
+            from_chat_id=chat_id,
+            message_id=media_message.message_id,
+            caption="<b>Test</b>",
+            parse_mode=ParseMode.HTML,
+            reply_to_message_id=media_message.message_id,
+            reply_markup=keyboard,
+        )
+        # we send a temp message which replies to the returned message id in order to get a
+        # message object
+        temp_message = bot.send_message(chat_id, "test", reply_to_message_id=returned.message_id)
+        message = temp_message.reply_to_message
+        assert message.chat_id == int(chat_id)
+        assert message.caption == "Test"
+        assert len(message.caption_entities) == 1
+        assert message.reply_markup == keyboard
+
+    """#TODO: Add to test
+        ({'allow_sending_without_reply': True}, None),
+        ({'allow_sending_without_reply': False}, None),
+        ({'allow_sending_without_reply': False}, True),
+    """
+
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    @pytest.mark.parametrize(
+        'default_bot,custom',
+        [
+            ({'parse_mode': ParseMode.HTML}, None),
+            ({'parse_mode': False}, None),
+            ({'parse_mode': False}, True),
+        ],
+        indirect=['default_bot'],
+    )
+    def test_copy_message_with_default(self, default_bot, chat_id, media_message, custom):
```

Comment:
I don't really understand what the `custom` parameter does here … like what's the difference between the 2nd and the 3rd parameter combination?

Reply:
I copied that from somewhere else. Just gonna remove it then, didnt make much sense to me as well, but hey, I am just testing

Reply:
then please also remove the parameter alltogether to avoid future confusion ;)

Reply:
I dont see what you mean. The custom argument of the function? Its no in the code anymore

Reply:
oh, was probably looking at the old diff. sorry for the confusion